src/main/java/com/google/sps/utils/GameUtils.java

```diff
@@ -0,0 +1,43 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import com.google.appengine.api.datastore.Entity;
+import java.util.ArrayList;
+
+public final class GameUtils {
+
+  /**
+  * When a user creates or joins a game, checks to see whether the user already has a game by that name, and returns false if so
+  */
+  public boolean IsValidGameName(String gameName, Entity userEntity) {
+    
+    if(userEntity == null || gameName == ""){
+        return false;
+    }
+    
+    // list of game names for the given user
+    ArrayList<String> names = (ArrayList<String>) userEntity.getProperty("gameNames");
```

Comment:
(No action required for this PR, but please read)

Eventually you might want helper functions/classes to convert datastore Entities into something easier to work with. Converting Entity into a more specific User class would help you benefit from Java's type system.

If you keep using Entity as you are doing, the following bugs are possible:

* userEntity.getProperty("GameNames");  // misspelled, still compiles. Always an empty list? Or cast fails at runtime?
* IsValidGameName(name, gameEntity);  // gameEntity is an Entity but has the wrong data; compiles, no errors, but always returns True.

But if you convert Entity to/from a different class, then Java's type system will help you catch the bugs and write simpler code. The examples would look like this:

* user.getGaemNames();  // misspelled, won't compile. Also, no casting required.
* IsValidGameName(name, game);  // won't compile because `game` is not a `User`.

Reply:
I'll save this and talk to Liyah and Hannah about it